cpp/include/milvus-storage/filesystem/s3/s3_fs.h

```diff
@@ -50,9 +51,15 @@ class S3FileSystemProducer : public FileSystemProducer {
       options.region = std::getenv("REGION");
     }
 
+    if (std::getenv("PART_SIZE") != nullptr) {
```

Comment:
The part size can be a variable instead of a static configuration.

The part size is bounded by i) cloud service limits, e.g., for Amazon s3 the range is (5MB, 5GB), ii) max part size that guarantees the upload is safe from network failure. Within the bounds, the part size should leverage all available memory to reduce the number of write operations.